spatialdata/_core/_spatialdata_ops.py

```diff
@@ -281,3 +288,165 @@ def _describe_paths(paths: list[list[Union[int, str]]]) -> str:
             transformations.append(g[path[i]][path[i + 1]]["transformation"])
         sequence = Sequence(transformations)
         return sequence
+
+
+##
+def concatenate_tables(tables: list[AnnData]) -> Optional[AnnData]:
+    """
+    Concatenate a list of tables using AnnData.concatenate() and preserving the validity of region, region_key and instance_key
+
+    Parameters
+    ----------
+    tables
+        A list of tables to concatenate
+
+    Returns
+    -------
+    A table with all the tables concatenated. If the list of tables is empty, None is returned.
+
+    Notes
+    -----
+    Not all tables can be merged, they need to have compatible region, region_key and instance_key values. This function
+    checks this and merges if possible.
+
+    """
+    if len(tables) == 0:
+        return None
+    if len(tables) == 1:
+        return tables[0]
+
+    MERGED_TABLES_REGION_KEY = "annotated_element_merged"
+    MAX_CONCATENTAION_TABLES = 100
+    for i in range(MAX_CONCATENTAION_TABLES):
+        if i == 0:
+            key = MERGED_TABLES_REGION_KEY
+        else:
+            key = f"{MERGED_TABLES_REGION_KEY}_{i}"
+
+        all_without = True
+        for table in tables:
+            if key in table.obs:
+                all_without = False
+                break
+        if all_without:
+            MERGED_TABLES_REGION_KEY = key
+            break
```

Comment:
I'm missing what's this code suppose to be doing, could you please elaborate?

Reply:
Now also added a comment in the code. Pasting it here:
```
    # 1) if REGION is a list, REGION_KEY is a string and there is a column in the table, with that name, specifying
    # the "regions element" each key is annotating; 2) if instead REGION is a string, REGION_KEY is not used.
    #
    # In case 1), we require that each table has the same value for REGION_KEY (this assumption could be relaxed,
    # see below) and then we concatenate the table. The new concatenated column is correcly annotating the rows.
    #
    # In case 2), we require that each table has no REGION_KEY value. In such a case, contatenating the tables would
    # not add any "REGION_KEY" column, since no table has it. For this reason we add such column to each table and we
    # call it "annotated_element_merged". Such a column could be already present in the table (for instance merging a
    # table that had already been merged), so the for loop before find a unique name. I added an upper bound,
    # so if the user keeps merging the same table more than 100 times (this is bad practice anyway), then we raise an
    # exception.
    #
    # Final note, as mentioned we could relax the requirement that all the tables have the same REGION_KEY value (
    # either all the same string, either all None), but I wanted to start simple, since this covers a lot of use
    # cases already.
```